src/containers/UserToolkit/index.js

```diff
@@ -0,0 +1,283 @@
+
+import lawMakerStateBranch from '../../state/members-candidates';
+import selectionStateBranch from '../../state/selections';
+import townHallStateBranch from '../../state/townhall';
+import {
+  getUserMOCs,
+  getSelectedMember,
+  getSelectedMemberLinks,
+  getUid,
+  getUserName,
+  getSelectedLink,
+} from '../../state/user/selectors';
+import {
+  addMemberLink,
+  editMemberLink,
+  deleteMemberLink,
+  setSelectedLink,
+  getSelectedMemberInfo,
+} from '../../state/user/actions';
+
+const updateSuccess = () => {
+  message.success('Link Updated');
+}
+
+const success = () => {
+  message.success('Thanks for submitting info!', 4);
+};
+
+class UserToolkit extends React.Component {
+  constructor(props) {
+    super(props);
+    this.renderMocBtns = this.renderMocBtns.bind(this);
+    this.showModal = this.showModal.bind(this);
+    this.handleCancel = this.handleCancel.bind(this);
+    this.handleCreate = this.handleCreate.bind(this);
+    this.saveAddFormRef = this.saveAddFormRef.bind(this);
+    this.saveEditFormRef = this.saveEditFormRef.bind(this);
+    this.selectMoc = this.selectMoc.bind(this);
+    this.handleDeleteLink = this.handleDeleteLink.bind(this);
+    this.handleAutoFillMember = this.handleAutoFillMember.bind(this);
+    this.renderMemberLinks = this.renderMemberLinks.bind(this);
+    this.showConfirm = this.showConfirm.bind(this);
+    this.handleNoEventSubmit = this.handleNoEventSubmit.bind(this);
+    this.state = {
+      addVisible: false,
+      editVisible: false,
+    }
+  }
+
+  selectMoc(moc) {
+    this.props.getSelectedMemberInfo(moc)
+  }
+
+  handleAutoFillMember(govId) {
+    this.props.requestPersonDataById('mocData', govId)
+  }
+
+  showConfirm() {
+    this.props.requestPersonDataById('mocData', this.props.selectedMoc.govtrack_id).then(() => {
+      let member = this.props.currentTownHall.Member;
+      let submit = this.handleNoEventSubmit;
+      if (!member) { return; }
+      confirm({
+        title: `Submit No Events for ${member}?`,
+        onOk() {
+          return new Promise((resolve, reject) => {
+            submit();
+            setTimeout(resolve, 1000);
+          });
+        },
+        onCancel() { },
+      });
+    })
+  }
+
+  resetAll() {
+    const {
+      resetAllData,
+    } = this.props;
+    resetAllData();
+  }
+
+  handleNoEventSubmit() {
+    const {
+      currentTownHall,
+      submitMetaData,
+      peopleDataUrl,
+      userDisplayName,
+      uid,
+    } = this.props;
+    const metaData = {
+      eventId: currentTownHall.eventId,
+      govtrack_id: currentTownHall.govtrack_id || null,
+      mocDataPath: peopleDataUrl,
+      thp_id: currentTownHall.thp_id || null,
+      memberId: currentTownHall.govtrack_id || currentTownHall.thp_id,
+      uid,
+      userDisplayName,
+    };
+    currentTownHall.meetingType = 'No Events';
+    submitMetaData(metaData);
+    success();
+    return this.resetAll();
+  }
+
+  showModal = (type, link) => {
```

Comment:
This is hard to follow, why are these functions defined outside of a class? This also seems like it could be broken up into a few components.

Reply:
I have components for LinksList, Member Buttons, Edit form and Create form. Would you recommend a component for displaying messages?

Reply:
I have added messages to class

Reply:
I mistakenly though these were outside the class since they were arrow functions. Please change them all to methods on the class for consistency 
Ie 
` showModal (type, link) {`